### ISSUE TITLE
Add reference to `processor.{event,type}` removal in changelog and update docs

### ISSUE THREAD

**Issue by marclop** [elastic/apm-server#11989]
In `8.10.0` onwards, we removed processor.event and processor.name from the model completely (https://github.com/elastic/apm-data/pull/79). Now, these are set by Elasticsearch in ingest pipelines: https://github.com/elastic/apm-server/pull/11028. This all works well for the majority of use cases.

However, it appears we overlooked or dismissed use-cases where we users use those fields in Logstash for routing purposes (https://www.elastic.co/guide/en/apm/guide/current/logstash-output.html#ls-config-pipeline)

![image](https://github.com/elastic/apm-server/assets/7286993/b4b2a7d4-4c43-4f59-841f-5828d8b5cc3a)

Since those fields are no longer set within APM Server, users can't use those fields' values in Logstash to route events. However, they could use `[data_stream][type]` instead of `[processor][event]` to route `traces`, `metrics` and `logs` to a datastream if they wanted to.

**Comment by bmorelli25** [comment]
@colleenmcginnis, there might be some relation to the changes made in https://github.com/elastic/apm-server/pull/11165. Not sure, but posting so I don't forget.

**Comment by bmorelli25** [comment]
Ah yes. @marclop, I was able to get the following config to work in 8.10 (which includes using `[processor][event]`. See this thread: https://github.com/elastic/apm-server/pull/11165#discussion_r1263094037.

```
output {
  if [@metadata][beat] == "apm-server" {
    if [processor][event] == "metric" {
      elasticsearch {
        index => "%{[data_stream][type]}-custom-%{[data_stream][namespace]}"
        action => "create"
        cloud_id => "${CLOUD_ID}"
        cloud_auth => "${CLOUD_AUTH}"
      }
    } else {
      elasticsearch {
        data_stream => "true"
        cloud_id => "${CLOUD_ID}"
        cloud_auth => "${CLOUD_AUTH}"
      }
    }
  }
}
```

**Comment by marclop** [comment]
@bmorelli25 it seems we only merged the changes in APM Server on July 18th. The documentation PR was tested on July 14th. Which means that unless you've re-tested the configuration above with a released `8.10.x` build, it's highly unlikely it'll work.

<img width="881" alt="image" src="https://github.com/elastic/apm-server/assets/7286993/dbe10b27-8949-4df0-bf51-46c9c2272687">
<img width="801" alt="image" src="https://github.com/elastic/apm-server/assets/7286993/0cf37458-a3a8-4048-84d0-1385426a8009">

**Comment by bmorelli25** [comment]
Ahh. My apologies for not checking the merge dates. That makes sense. Thanks!
